### ISSUE TITLE
Radio support

### ISSUE THREAD

**Issue by mazkolain-zz** [mazkolain-zz/spotimc#17]
Include the radio feature, like the official desktop client does.

**Comment by bass000** [comment]
Can I "+1" this one, it would be great!

**Comment by chrstnwhlrt** [comment]
+1

**Comment by azgul** [comment]
would be baller as fuck. +1

**Comment by makomatic** [comment]
+1

**Comment by makomatic** [comment]
It should be possible to implement an own radio function with the help of the echonest api. I'll try and share my tests...

**Comment by mazkolain-zz** [comment]
My plan is to implement radio using the search subsystem, but setting the search type to "suggest":
https://developer.spotify.com/technologies/libspotify/docs/12.1.45/group__search.html#ga2f64dfbcc1313e7d5179998cb28208d4

However, I've never tested this for myself.

**Comment by makomatic** [comment]
I've already found this part:
http://developer.echonest.com/docs/v4/playlist.html#dynamic-create
I'm quite sure that this is the way spotify does is - it even returns the same similar artists/tracks.
You are also able to retrieve spotify track id's or even the whole playlist as a spotify parseble id.

-> as I'm using python for this one, I play with http://echonest.github.com/pyechonest/playlist.html at the moment - quite fascinating

**Comment by mazkolain-zz** [comment]
@makomatic A little drawback I noticed if it's going to be implemented through Echonest:

Currently the Echonest api has a limit of 120 requests per minute (for non-commercial users):
http://developer.echonest.com/docs/v4/index.html#rate-limits

For standalone clients this is just fine, but this limit is shared accross all the clients using the same api key, so I would expect constant rate limiting errors there.

**Comment by makomatic** [comment]
@mazkolain Yeah, I know - an own api key is needed with this solution - you can easily register there.
But anyway, I've no time to work on this idea at the moment ;)

**Comment by screendriver** [comment]
+1

**Comment by d0p3fish** [comment]
+1

**Comment by marminthibaut** [comment]
+1

**Comment by safts** [comment]
+1, radio is amazing, it's the easiest way to discover music in Spotify
